### PR TITLE
Spawn Zombie Cops with small firearms more often and lower pick rate of mx_military 'map extra'

### PR DESCRIPTION
**Pull request by KurzedMetal**
#### Summary
Balance "Spawn Zombie Cops with small firearms more often and lower pick rate of mx_military 'map extra'"

#### Purpose of change

The intention is to make the player have less fire power early in the game and reducing overall progress into mid-late game firepower.

This will be done by reducing drops on specific map encounters that generates too much loot with low risk and reduce the general drop rate of good quality rifles and their ammo from zombie soldiers.
In particular I'm going to target generation of M4 carbines and 5.56 NATO ammo.

After this change the players will probably have to be lucky to get a pristine firearm from zombie soldiers (or fix them themselves) and will have to be more conservative with the ammo.

#### Describe the solution

I made multiple small changes in different places:
* **Changed Zombie Cops firearm drops: before this change 10 cops would generate 2 rifles and 2 handguns, now they generate 1 rifle and 4 handguns.** Reasoning: It's more common for the police to have small arms.
* **Reduced pick rate of mx_military "map extra"** by a factor of 5 (the nether creature + soldiers with blood + weapons on the ground)
* ~**Zombie soldiers will drop a max of 20 ammo instead of 30 now (mag never full)**. Reasoning: A soldier would probably never go down without using resources at their disposal. Intention: Reduce ammo~ - **Rejected by devs**
* ~**Reduce grenades generated per Soldier, reduced from 1-3 to 1-2**. Same reasoning as before.~ - **Rejected by devs**
* ~**Reduced the generation of rifles on the ground on two specific "map extras"** (mx_roadblock and mx_military) from 90% per corpse to 25% per corpse. **These events would usually generated several rifles in pristine state with quite a bunch of ammo.** In particular mx_military which is the scenario with 1 Nether creature + several Soldiers is quite easy to loot without combat (soldiers chase the Neather creature away). After this change, it will generate only 1 rifle on the group per event.~ - **Rejected by devs**

#### Describe alternatives you've considered

I was also interested in reduced ammo dropped by turrets, which is quite ridiculous right now.
But I wouldn't know where to start and also there seem to be at least an issue tracking that: #53897

#### Testing

I actually haven't tested the changes yet, but they are very simple to review as I didn't add/remove too much, just tweaked a few numbers here and there after analyzing how they were defined.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
@kevingranade is wolking on turrets right now, I believe.

> Zombie soldiers will drop a max of 20 ammo instead of 30 now (mag never full). Reasoning: A soldier would probably never go down without using resources at their disposal. Intention: Reduce ammo

This feels quite arbitrary to me - it is reasonable for soldiers to die with full mags.

> Reduced the generation of rifles on the ground on two specific "map extras" (mx_roadblock and mx_military) from 90% per corpse to 25% per corpse. These events would usually generated several rifles in pristine state with quite a bunch of ammo. In particular mx_military which is the scenario with 1 Nether creature + several Soldiers is quite easy to loot without combat (soldiers chase the Neather creature away). After this change, it will generate only 1 rifle on the group per event.

I think this is wrong - I don't like vanishing things like this, but I do think this map extra should be _significantly_ rarer. Also, adjustments to fix the 'chasing after nether monster' would be another good route - I'm not sure there's a whole lot of reason for nether mons to be hanging around bodies like that.

File: data/json/monsterdrops/zombie_soldier.json [hunk 1]
```diff
@@ -14,14 +14,14 @@
       },
       {
         "distribution": [
-          { "group": "military_standard_assault_rifles", "contents-item": "shoulder_strap", "prob": 75 },
-          { "group": "military_standard_lmgs", "contents-item": "shoulder_strap", "prob": 10 },
-          { "group": "military_standard_shotguns", "contents-item": "shoulder_strap", "prob": 5 }
+          { "group": "military_standard_assault_rifles", "contents-item": "shoulder_strap", "prob": 75, "damage": [0, 4] },
+          { "group": "military_standard_lmgs", "contents-item": "shoulder_strap", "prob": 10, "damage": [0, 4] },
+          { "group": "military_standard_shotguns", "contents-item": "shoulder_strap", "prob": 5, "damage": [0, 4] }
```
Comment (anothersimulacrum) [inline data/json/monsterdrops/zombie_soldier.json:19]: Consider also adding fouling.

**Comment by bombasticSlacks** [comment]
Yeah standard us soldiers iirc carry 9 magazine on them. I don't think going from 30 to 20 when the actual number should be between 270 and 0 makes too much sense.

**Comment by KurzedMetal** [comment]
That would make sense only if they didn't fire any shot at all.
Which I don't think it would happen in a zombie apocalypse, and definitely not before dying.

**Comment by bombasticSlacks** [comment]
Right my point is if this change is "for realism" where did their empty mags go? Why do they never die with any spare magazines on them? It would be way more likely for them to after death hold onto whatever mags they hadn't fired since they would be affixed to them, even if they drop / lose their rifle.

**Comment by anothersimulacrum** [comment]
I disagree, thesebsoldiers aren't just dying from fighting zombies (ambushed by nether creatures, turning on eachother, etc), and it is entirely reasonable for soldiers to die with a full magazine in their gun.

File: data/json/monsterdrops/zombie_soldier.json (hunk 1 above)
Comment (KurzedMetal) [reply data/json/monsterdrops/zombie_soldier.json]: The attribute for fouling is "dirt", right?
Do you suggest any particular value range for it?

Comment (KurzedMetal) [reply data/json/monsterdrops/zombie_soldier.json]: Added from 0 to 6000 for now, if you think another value range would be better feel free to mention it.

Comment (anothersimulacrum) [reply data/json/monsterdrops/zombie_soldier.json]: I don't have suggestions for a range, and I did not remember the key for fouling (and couldn't check at the time), so changes seem good.

**Review by ymber — CHANGES_REQUESTED** [review]
The cop sidearms and fouling can stay, everything else in this PR has to go. We've already done this at least once. These things were set the way they are for a reason.  #36645

**Comment by Inglonias** [comment]
Also see #53245 for more discussion on this.

**Comment by KurzedMetal** [comment]
I'm ok letting the ammo & grenade change go.

But I started this PR mostly for the mx_military scenario, it basically trivializes early game giving the player so much fire power.
I'm an experienced played, and I can tell you that most of the time I start a new character my first firearms come from mx_military when I start exploring the world with my first vehicle.

You just have to find it at day and wait until the Nether spawn drive the soldiers away and there you have a weapon per soldier waiting for you for free in the ground.

If my proposed solution to the problem doesn't work, I'd like to see an alternative solution:
* Would significantly reducing the chance to spawn the mx_military scenario be accepted like @anothersimulacrum suggested? This scenario would still be a loot piñata but at least it would be rare and feel rewarding.
* Or can we remove the Nether spawn and force the player to deal with the soldiers? (This would probably still make it easy to for the player to lure them away with a vehicle and come back for the loot).
* I'd really like to change this somehow, I'm accepting alternatives proposed if the scope isn't too big.

**Comment by Inglonias** [comment]
I agree that the mx_military extra is definitely the biggest offender here. If you want some math, there's a spreadsheet to play with in the issue I linked above that tells you roughly how many of them you'll find in various overmap tiles. Toning that down and/or changing it, combined with #55790 is probably for the best.

Suggestions on how to do that - Replace a few of the zombie soldiers with feral soldiers. This would drastically reduce the day 1 problem while still keeping these extras rewarding. Consider making a variant of the zombie and feral soldiers that spawn in that extra and only that extra so you have better control over their loot drops. Give them the same name, sprite, and description as their non-extra counterparts, but different loot tables.

Get rid of the guns on the ground completely. Put those guns in the hands or backpacks of the zombies. This is where the monster variants come in. We don't want every single zombie soldier or feral soldier spawning with an M4 or M16, but we want these ones to have some. Spawn those weapons jammed or damaged to alleviate the day 1 advantage of this extra some more. I feel that the ammo supply from mx_military is ok, but someone more senior than me is welcome to add their own thoughts.

**Comment by KurzedMetal** [comment]
For now I reduced the chance to spawn mx_military, if the devs have any other change requests, feel free to mention them.

**Comment by Inglonias** [comment]
I'm actually working on a branch right now with the changes I proposed above - To recap:

* New mx_military specific variants of the zombie soldier and feral soldier. The feral soldier can fire a few shots from a handgun and make up one in three of the total soldiers present.
* Guns on the ground are gone. Instead, the mx_military variants of the zombies will drop them.

**Comment by Inglonias** [comment]
I'm linking to what I would do to change up the mx_military extra. This is code you can feel free to steal if you think it's a good idea:

[Changes mentioned above](https://github.com/Inglonias/Cataclysm-DDA/commit/5d44237ddd47ba61dc27e742b05c0dff9401e5f7)
[I like your itemgroups, so I stole them](https://github.com/Inglonias/Cataclysm-DDA/commit/83930523dd7efb53425d6485ce859403c4963629)

**Comment by anothersimulacrum** [comment]
I am not a big fan of duplicating monsters like that does.

**Comment by Inglonias** [comment]
> I am not a big fan of duplicating monsters like that does.

Be that as it may, there is a precedent for doing that. We have three copies of feral humans with different weapons.

**Comment by anothersimulacrum** [comment]
Weapons also change their damage output - I do not think adding new exact copies for a slight change to a map extra is worth it when there are other avenues to explore.
Additionally, it is more fine if the weapon ferals diverge a bit than if these soldiers do.

**Comment by KurzedMetal** [comment]
I liked your idea of increased the weight of the other extras to reduce mx_military.
Now both road and field spawns for mx_military are 5 times less likely than before, I think that should make it rare enough for now.

BTW, discuss the feral changes in the other PR instead of here?

**Comment by Inglonias** [comment]
You are correct, and I apologize for derailing your PR. Any further discussion will happen in the right place.

**Comment by Inglonias** [comment]
A factor of five is very low. I play with a mod that cuts the spawn rate in half from their defaults, and I have yet to see more than one or two across many, many overmaps.

Suggest you lower the spawn rate by something more like 25 - 33% rather that cutting it by five.

**Comment by ymber** [comment]
Cutting the spawn rate of that map extra is acceptable as is broadly reworking it. It's been around a very long time and its environmental storytelling value is pretty thin. Adding feral soldiers in some form is a good idea but how they fit with the story of their spawn location needs to be thought through.

One gun per soldier has to be maintained and if they're not carrying them on slings they'd have been dropped on the ground. Evaporating guns was a problem we more or less solved by setting these kind of spawns the way they are now. Zombie soldiers should actually be carrying significantly more ammo than they are now, they're missing mag pouches that should be there. Cutting grenade spawns may be a reasonable change, I don't know current US doctrine on grenades and how many they'd be carrying in the scenario.

It is acceptable for events that the player was uninvolved in to offer benefits to the player. If mi-go behavior that makes sense in the setting leaves a load of military gear available to the player that's fine. That said I'd be inclined to rework the extra into something more fleshed out and add some kind of feral soldiers. More depth and more challenge to get the gear wouldn't be a bad thing.

**Comment by bombasticSlacks** [comment]
> they're missing mag pouches that should be there

Getting a bit off topic but I've actually added Load Bearing Vests and soldiers have a chance to spawn with them and a number of MOLLE compatible pouches for their vests. The pouches could be populated, so if a soldier has a flashlight pouch it has a flashlight, same with a radio, same with 2 30 rounders in a mag pouch.

File: data/json/regional_map_settings.json [hunk 2]
```diff
@@ -523,55 +523,55 @@
...
-          "mx_pond": 20,
-          "mx_point_burned_ground": 50,
-          "mx_casings": 20,
-          "mx_corpses": 3,
-          "mx_nest_wasp": 5,
-          "mx_mass_grave": 5,
-          "mx_grave": 5,
-          "mx_exocrash_1": 1,
-          "mx_exocrash_2": 1
+          "mx_helicopter": 10,
```
Comment (anothersimulacrum) [inline data/json/regional_map_settings.json:526]: This should stay less frequent than mx_military.

Comment (KurzedMetal) [reply data/json/regional_map_settings.json:526]: what values do you suggest then?

Comment (anothersimulacrum) [reply data/json/regional_map_settings.json:526]: 2 would keep it the same ratio, but 1 is also acceptable.

**Pull request merged**